openmc/mgxs_library.py

```diff
@@ -553,23 +585,526 @@ def nu_fission(self, nu_fission):
                 msg = "Invalid Shape of Nu_fission!"
                 raise ValueError(msg)
         else:
-            # Get shape of nu_fission so we can figure if we need chi or not
+            # Get shape of nu_fission to determine if we need chi or not
             if nu_fission.shape == shape_vec:
                 self._use_chi = True
-                shape = shape_vec
             elif nu_fission.shape == shape_mat:
                 self._use_chi = False
-                shape = shape_mat
             else:
                 msg = "Invalid Shape of Nu_fission!"
                 raise ValueError(msg)
 
         # check we have a numpy list
-        check_type("nu_fission", nu_fission, np.ndarray, expected_iter_type=Real)
+        check_type("nu_fission", nu_fission, np.ndarray,
+                   expected_iter_type=Real)
         self._nu_fission = np.copy(nu_fission)
         if np.sum(self._nu_fission) > 0.0:
             self._fissionable = True
 
+    def set_total(self, total, **kwargs):
```

Comment:
The docstring should make it clear that if `total` is an `ndarray`, a *deep copy* of the array is made during the assignment. Sophisticated users might naively assume they can take advantage of array manipulations between simulations without (re-)assigning the array to the `XSdata` object.